### PR TITLE
[Feature] Be able to select option with Return key

### PR DESCRIPTION
**Pull request by faizanu94**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number: https://github.com/reaviz/reaselct/issues/3

## What is the new behavior?
Now user will be able to select option with return key

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/Select.tsx [hunk 1]
```diff
@@ -340,9 +340,9 @@ export const Select: FC<Partial<SelectProps>> = ({
       }
 
       if (index > -1 || createable) {
-        const newSelection = result[index]?.value || {
-          value: inputValue,
-          children: inputValue
+        const newSelection = {
```
Comment (amcdnl) [inline src/Select.tsx:343]: Wow - this was it? Was it just passing the wrong var?

Comment (faizanu94) [reply src/Select.tsx:343]: yup, I verified through storybook :)

**Comment by amcdnl** [comment]
released!

**Comment by amcdnl** [comment]
@faizanu94  I think this caused a regression with tagging.

**Comment by faizanu94** [comment]
@amcdnl yeah I just saw a fix you merged with `createable` flag used

**Pull request merged**